CHIMERA_V2.py

```diff
@@ -0,0 +1,215 @@
+from astropy import wcs
+from astropy.io import fits
+from astropy.modeling.models import Gaussian2D
+import astropy.units as u
+from astropy.visualization import astropy_mpl_style
+import glob
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy
+import scipy.interpolate
+import sunpy
+import sunpy.map
+import sys
+import sunpy.data.sample
+
+
+plt.style.use(astropy_mpl_style)
+
+# loading in the images as fits files
+
+im171 = glob.glob('171.fts')
+im193 = glob.glob('193.fts')
+im211 = glob.glob('211.fts')
+imhmi = glob.glob('hmi.fts')
+
+#ensure that all images are present
+
+if im171 == [] or im193 == [] or im211 == [] or imhmi == []:
+	print("Not all required files present")
+	sys.exit()
+
+#Two functions that rescale the aia and hmi images from any original size to any final size
+
+#didn't normalize by exposure time for hmi because it was equal to 0
+
+def rescale_aia(image: np.array, orig_res: int, desired_res: int):
+        hdu_number = 0
+        hed = fits.getheader(image[0],hdu_number)
+        dat= fits.getdata(image[0], ext=0)/(hed["EXPTIME"])
+        if desired_res > orig_res:
+            scaled_array=np.linspace(start = 0, stop = desired_res, num = orig_res)
+            dn=scipy.interpolate.RectBivariateSpline(scaled_array,scaled_array,dat)
+            if len(dn(np.arange(0, desired_res),np.arange(0,desired_res))) != desired_res:
+                print("Incorrect image resolution")
+                sys.exit()
+            else:
+                return dn(np.arange(0,desired_res),np.arange(0,desired_res))
+        elif desired_res < orig_res:
+            scaled_array=np.linspace(start = 0, stop = orig_res, num = desired_res)
+            dn=scipy.interpolate.RectBivariateSpline(scaled_array,scaled_array,dat)
+            if len(dn(np.arange(0, desired_res),np.arange(0,desired_res))) != desired_res:
+                print("Incorrect image resolution")
+                sys.exit()
+            else:
+                return dn(np.arange(0,desired_res),np.arange(0,desired_res))
```

Comment:
Have a look at sunpy reproject it does this but with some additional checks etc

Reply:
I checked it out but I couldn't figure out how to get it to work with the rest of the code. The way I rescaled it should work as it's the same as how the old chimera code did it